lumicks/pylake/scan.py

```diff
@@ -91,10 +106,146 @@ def _plot(self, image, frame=1, **kwargs):
             aspect=(image.shape[0] / image.shape[1]) * (x_um / y_um)
         )
 
-        plt.imshow(image, **{**default_kwargs, **kwargs})
+        if not image_handle:
+            image_handle = plt.imshow(image, **{**default_kwargs, **kwargs})
+        else:
+            # Updating the image data in an existing plot is a lot faster than re-plotting with `imshow`.
+            image_handle.set_data(image)
+
         plt.xlabel(f"{axis_label[axes[0]['axis']]} ($\\mu m$)")
         plt.ylabel(f"{axis_label[axes[1]['axis']]} ($\\mu m$)")
         if self.num_frames == 1:
             plt.title(self.name)
         else:
             plt.title(f"{self.name} [frame {frame}/{self.num_frames}]")
+
+        return image_handle
+
+    def _export_video(self, plot_type, file_name, start_frame, end_frame, fps, **kwargs):
+        """Export a video of a particular scan plot
+
+        Parameters
+        ----------
+        plot_type : str
+            One of the plot types "red", "green", "blue" or "rgb".
+        file_name : str
+            File name to export to.
+        start_frame : int
+            First frame in exported video.
+        end_frame : int
+            Last frame in exported video.
+        fps : int
+            Frame rate.
+        **kwargs
+            Forwarded to :func:`matplotlib.pyplot.imshow`."""
+        from matplotlib import animation, rcParams
+        import matplotlib.pyplot as plt
+
+        metadata = dict(title=self.name)
+        if "ffmpeg" in animation.writers:
+            writer = animation.writers["ffmpeg"](fps=fps, metadata=metadata)
+        elif "pillow" in animation.writers:
+            writer = animation.writers["pillow"](fps=fps, metadata=metadata)
+        else:
+            raise RuntimeError("You need either ffmpeg or pillow installed to export videos.")
+
+        start_frame = start_frame if start_frame else 1
+        end_frame = end_frame if end_frame else self.num_frames + 1
+
+        # On some notebook backends, figures render with a transparent background by default. This leads to very
+        # poor image quality, since it prevents font anti-aliasing (at the cost of not having transparent regions
+        # outside the axes part of figures).
+        face_color = rcParams["figure.facecolor"]
+        face_color_rgba = to_rgba(face_color)
```

Comment:
You imported all `matplotlib` bits at the function level, except this one. Any reason for the difference?

Reply:
Yeah, fair point. I moved it to the function scope, since we really only use that functionality there and this function is likely used rarely.

Do we have any standard guidelines for imports?

Reply:
For me, there are two possible reasons to put imports at function-level instead of module-level:
1. If it's an optional dependency -- the user will only get the import error if they try to use the feature that requires it.
2. If the module load time is long we may want to defer its initialization until someone actually uses the feature. I think that I noticed this with `scipy` or `matplotlib` on Windows. Although, with our use case, I think we almost always need `matplotlib` anyway. Not a super important point unless there's some clear data to indicate slowdowns.